collections/app/controllers/CollectionsController.scala

```diff
@@ -126,11 +137,35 @@ object CollectionsController extends Controller with ArgoHelpers {
     } getOrElse Future.successful(invalidJson(req.body))
   }
 
+  type MaybeTree = Option[Node[Collection]]
+  def hasChildren(path: List[String]): Future[Boolean] =
+    allCollections.map { tree =>
+
+      // Traverse the tree using the path
+      val maybeTree = path
+        .foldLeft[MaybeTree](Some(tree))((optBranch, nodeName) => {
+
+          optBranch.flatMap { _.children.find(_.basename == nodeName) }
+        })
+
+      // Does out target node have children?
+      !maybeTree.flatMap(_.children.headOption).isEmpty
```

Comment:
intelliJ is suggesting isDefined vs !isEmpty, might be a little clearer but doesn't matter really